### PR TITLE
Lazily fill out the ItemList for network delta updates

### PR DESCRIPTION
**Pull request by eigenraven**
Attempt at improving the performance of endgame bases, this code should be equivalent in terms of behaviour. I tested it on a couple of simple ae2 contraptions and I couldn't find anything wrong in a few minutes of tests, but it'd have to be tested on a full base to be sure.

Jenkins build: http://jenkins.usrv.eu:8080/job/Applied-Energistics/706/

### CODE REVIEW

**Review by repo-alt — DISMISSED** [review]

**Comment by eigenraven** [comment]
This breaks storage bus "report inaccessible items" reporting, I'll explore alternative solutions

**Comment by eigenraven** [comment]
I have now introduced an additional API for the purpose of these single-item lookups and put optimized implementations in every class in AE2 that can have one. I tested it and it seems to now also work with the "report inaccessible items" setting.

File: src/main/java/appeng/api/storage/IMEInventory.java [hunk 1]
```diff
@@ -50,13 +50,26 @@ public interface IMEInventory<StackType extends IAEStack> {
...
+    /**
+     * Request a report of how many of a single item type are available in storage. It falls back to
+     * {@link IMEInventory#getAvailableItems} if a more optimized implementation is not present.
+     *
+     * @param request The item type to search for, it does not get modified
+     * @return A new stack with the stack size set to the count of items in storage, or null if none are present
+     */
+    @SuppressWarnings("unchecked") // changing the generic StackType to be correct here is too much of a breaking API
+                                   // change
+    default StackType getAvailableItem(StackType request) {
```
Comment (firenoo) [inline src/main/java/appeng/api/storage/IMEInventory.java:69]: nit: This should probably be annotated with `@Nullable` or equivalent

**Review by firenoo — COMMENTED** [review]
Code logic wise, looks good

File: src/main/java/appeng/api/storage/IMEInventory.java (hunk 1 above)
Comment (eigenraven) [reply src/main/java/appeng/api/storage/IMEInventory.java]: I added NonNull to the request, Nullable is the default in ae2 (and most mc mods) as there is no package-level override.

**Review by repo-alt — APPROVED** [review]
that may work, let someone test that on a big base

**Comment by S4mpsa** [comment]
We have tested this in Ballers base and haven't found any problems.

**Pull request merged**